### PR TITLE
feat(linux): use new km_core_keyboard_load_from_blob API method 🎼

### PR DESCRIPTION
**Pull request by ermshiperete**
Fixes: #12498

# User Testing

**TEST_WORKS**:  Keyman keyboards should still work as before, i.e. switch to a Keyman keyboard and verify that it behaves as expected.

### CODE REVIEW

File: linux/ibus-keyman/src/engine.c [hunk 1]
```diff
@@ -516,7 +516,33 @@ ibus_keyman_engine_constructor(
...
-    status = km_core_keyboard_load(abs_kmx_path, &(keyman->keyboard));
+    FILE* kmx_file = fopen(abs_kmx_path, "rb");
+    if (!kmx_file) {
+      g_warning("%s: problem opening kmx_file %s.", __FUNCTION__, abs_kmx_path);
+      return NULL;
+    }
+
+    fseek(kmx_file, 0, SEEK_END);
+    long length = ftell(kmx_file);
+    fseek(kmx_file, 0, SEEK_SET);
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/engine.c:527]: No error checking?

**Review by mcdurdin — APPROVED** [review]
LGTM

File: linux/ibus-keyman/src/engine.c (hunk 1 above)
Comment (ermshiperete) [reply linux/ibus-keyman/src/engine.c]: ok ok. Done :smile:

**Review by mcdurdin — APPROVED** [review]

**Comment by dinakaranr** [comment]
# Test Results
I tested this issue with the attached keyman 18.0.138-alpha (package version
18.0.138-1~PR-12683-4025.1+focal1) build on the Jammy & Focal OS environment. Here is my observation.

* **TEST_WORKS (Passed):** 
1. Installed the Keyman package version 118.0.138-1~PR-12683-4025.1+focal1
2. Open the configuration dialog by running the "km-config" command in the terminal.
3. "Download Keyman keyboards" dialog by clicking the "Download keyboard" button.
4. Installed the SIL IPA keyboard.(search a keyboard in the search box)(Khmer, gff_Amheric)
5. Verified that the installed keyboard appears on the configuration dialog.
6. Open the Note editor & Libre Office application
7. Switch the keyboard from English to Khmer.
8. Enter some words in the application.
9. Switch the keyboard from Khmer to IPA.
10. Enter some words in the application.
11. Verified the text appeared in the application and the keyboard switch works well.
It works well with Jammy and Focal OSs. Thank you.

**Pull request merged**